lib/colorls/flags.rb

```diff
@@ -34,6 +36,52 @@ def process
 
     private
 
+    def add_sort_options(options)
+      options.separator ''
+      options.separator 'sorting options:'
+      options.separator ''
+      options.on('-sMODE', %w[d f], 'sort (f)iles or (d)irectories first') do |m|
+        @opts[:sort] = m == 'd' ? :dirs : :files
+      end
+      options.on('--sort-dirs', '--group-directories-first', 'sort directories first') { @opts[:sort] = :dirs }
+      options.on('--sort-files', 'sort files first')                                   { @opts[:sort] = :files }
+    end
```

Comment:
`-sTYPE show TYPE first, -sd = directories first, -sf = files first` seems better. 😄 

1. I'm not sure of how `optparse` implements TAB auto-complete, but does `-sd` & `-sf` auto-complete?
2. Maybe we should rename `-sf`, `-sd` to `-F`, `-D`? But this would break these tags for existing users, and I'm not quite with this idea.